r/tests/testthat/test-duckdb.R

```diff
@@ -15,6 +15,15 @@
 # specific language governing permissions and limitations
 # under the License.
 
+test_that("meaningful error message when duckdb is not installed", {
+  skip_if(requireNamespace("duckdb", quietly = TRUE))
+  ds <- InMemoryDataset$create(example_data)
+  expect_error(
+    to_duckdb(ds),
+    "Please install the `duckdb` package."
+  )
+})
+
```

Comment:
Would you mind adding some comments in here that describe some of the oddities of this test so we know why it's doing the strange but necesary things it is when we look at it in the future:

* The reason we skip when duckdb is installed
* The reason this test is above the rest of the tests

Reply:
Done.